src/components/generics/vacation-test-forms/myVacationRequests.tsx

```diff
@@ -11,43 +11,38 @@ import Holidays from "date-holidays";
 import strings from "localization/strings";
 import { FilterScopes } from "types";
 import DateRangePicker from "../date-range-picker/date-range-picker";
-
-/**
- * interface type request
- */
-interface Request {
-  id: number;
-  vacationType: string;
-  comment: string;
-  employee: string;
-  days: number;
-  startDate: string;
-  endDate: string;
-  status: string;
-}
+import { Request } from "types/index";
+import { VacationRequest, VacationType } from "generated/client";
+import { useAppSelector } from "app/hooks";
+import { selectPerson } from "features/person/person-slice";
+import Api from "api/api";
+import { selectAuth } from "features/auth/auth-slice";
+import { ErrorContext } from "components/error-handler/error-handler";
 
 /**
  * Renders vacation request table
  */
 const renderVacationRequests = () => {
   const classes = useEditorContentStyles();
+  const { person } = useAppSelector(selectPerson);
+  const { accessToken } = useAppSelector(selectAuth);
   const [ dateFormat ] = useState("yyyy.MM.dd");
   const [ selectedVacationStartDate, setSelectedVacationStartDate ] = useState(new Date());
   const [ selectedVacationEndDate, setSelectedVacationEndDate ] = useState(new Date());
   const [ openRows, setOpenRows ] = useState<boolean[]>([]);
-  const [ newTextContent, setNewTextContent ] = useState("");
-  const [ newVacationType, setNewVacationType ] = useState("");
+  const [ textContent, setTextContent ] = useState("");
+  const [ vacationType, setVacationType ] = useState<VacationType | string>();
   const [ datePickerView ] = useState<CalendarPickerView>("day");
-  const [ textContent ] = useState("");
-  const [ vacationType ] = useState("");
-
+  const context = useContext(ErrorContext);
+  const [ requests ] = useState<VacationRequest[]>([]);
```

Comment:
Same as in the previous file, this requests is being modified later on using .push, but it should be modified using the setRequests which should be declared here.

ALSO! This seems to have the potential for duplication, please check if its necessary to be using this both here and in the editor-content.tsx
Happy to chat about whats going on here.